pkg/envoy/store/compose.go

```diff
@@ -187,7 +187,17 @@ func (d *composeDecoder) decodeComposeRecord(ctx context.Context, s store.Storer
 			for _, f := range ff {
 				fNs := *f
 				fNs.NamespaceID = nsID
-				ffNs = append(ffNs, &fNs)
+
+				nn, _, err := s.SearchComposeModules(ctx, types.ModuleFilter{
+					NamespaceID: nsID,
+				})
+				if err != nil {
+					continue
+				}
+				for _, n := range nn {
+					fNs.ModuleID = n.ID
```

Comment:
Won't this only include the last module in the slice? Shouldn't we copy the `fNs`?

Reply:
yes you are right, it is fixed now